x-pack/plugins/triggers_actions_ui/public/application/components/rules_setting/rules_settings_modal.tsx

```diff
@@ -84,21 +104,24 @@ export const RulesSettingsModal = memo((props: RulesSettingsModalProps) => {
     },
   } = capabilities;
 
-  const [flappingSettings, setFlappingSettings] = useState<RulesSettingsFlappingProperties>();
-  const [hasFlappingChanged, setHasFlappingChanged] = useState<boolean>(false);
+  const [flappingSettings, hasFlappingChanged, setFlappingSettings, resetFlappingSettings] =
+    useResettableState<RulesSettingsFlappingProperties>();
 
-  const [queryDelaySettings, setQueryDelaySettings] = useState<RulesSettingsQueryDelayProperties>();
-  const [hasQueryDelayChanged, setHasQueryDelayChanged] = useState<boolean>(false);
+  const [queryDelaySettings, hasQueryDelayChanged, setQueryDelaySettings, resetQueryDelaySettings] =
+    useResettableState<RulesSettingsQueryDelayProperties>();
 
   const { isLoading: isFlappingLoading, isError: hasFlappingError } = useGetFlappingSettings({
```

Comment:
Could we use `data` from `useGetFlappingSettings` and `useGetQueryDelaySettings` as a source of previous values for settings?